### DISCUSSION TITLE
Add project scaffolding

### DISCUSSION THREAD

**Discussion by manekinekko** [Azure/static-web-apps-cli#384] [Ideas]
**Describe the solution you'd like**
As a user I'd like to be able to use `swa` to scaffold a new project structure.

Example: running `swa init` should produce the following project structure:

```
├── api
│   └── <azure functions project>
├── app
│   └── <front-end project>
└── swa.json
```

Other options/questions:
- add an interactive mode (like in https://hexa.run)?
- only scaffold the `api` folder?

**Comment by aaronpowell** [comment]
Unless it's for setting up the SWA config file (per #20) I feel like the implementation effort would greatly outweigh the user value.

We have a lot of permutations that we'd need to support, first there's the client framework (React, Vue, Angular, Svelte, Vanilla JS, Blazor), then whether you want an API, and if so, what's the API written in (JS, TypeScript, C#, F#, Python). Then there's static site generators (VuePress, Gatsby, Hugo, Jekyll, Nuxt) that would also be options.

**Comment by aaronpowell** [comment]
Thinking about this - would it be possible to use the GitHub repo templates to generate a project? That would make it easier to have the permutations handled (just create a new template)

**Comment by manekinekko** [comment]
I am in favor of this 👍

**Comment by anthonychu** [comment]
We need to think about what problem this addresses. Is the goal to prescribe an opinionated folder structure? GitHub templates might be a good idea but there's a maintenance overhead to them. How many people would find them useful?

We'll be doing some research studies in the next few months around the CLI. We'll include the getting started experience. /cc @Reshmi-Sriram

**Comment by aaronpowell** [comment]
> We need to think about what problem this addresses. Is the goal to prescribe an opinionated folder structure? GitHub templates might be a good idea but there's a maintenance overhead to them.

The goal would be to give people a starting point for creating a project, whether they are experienced with SWA or it's their first time. I think of it in a similar manner to someone using `func init` to setup the starting point for a Functions app.

**Comment by aaronpowell** [comment]
btw, I created a standalone tool to do this - https://dev.to/azure/scaffolding-static-web-apps-4p97
